src/lucky/routeable.cr

```diff
@@ -138,6 +197,30 @@ module Lucky::Routeable
     \{% end %}
   end
 
+  # Access a query parameter
+  #
+  # This will allow any action to accept and access query parameters. By
+  # default query parameters are ignored, but adding a `param` will make them
+  # available:
+  # ```
+  # class Posts::Index < BrowserAction
+  #   param page : Int32 = 1
+  #
+  #   action do
+  #     render_text "Posts - Page #{page}"
+  #   end
+  # end
+  # ```
+  #
+  # Visiting `/posts?page=10` would render the following:
+  #
+  # ```text
+  # Posts - Page 10
+  # ```
+  #
+  # Additionally, these parameters are typed. The path `/posts?page=ten` will
+  # raise a `Lucky::Exceptions::InvalidParam` error because `ten` is a string
+  # not an Int32.
```

Comment:
An example of generating a link would be good as well: `Posts::Index.with(2) # => "/posts?page=2"`

Maybe an example of a nilable para, as well?

Reply:
Oh! I didn't know you could do that with `with`

Reply:
Looking through the code, it looks like you can only use `with` to generate a query param if it's required. How would I generate a nillable param?

```crystal
query_params = {} of String => String
{% for param in PARAM_DECLARATIONS %}
  param_is_default_or_nil = {{ param.var }} == {{ param.value || nil }}
  unless param_is_default_or_nil
    query_params["{{ param.var }}"] = {{ param.var }}.to_s
  end
{% end %}
```